### PR TITLE
D3ASIM-3674: Enable Savings KPI restoration

### PR DESCRIPTION
**Pull request by faizan2590**

### CODE REVIEW

File: gsy_framework/sim_results/kpi.py [hunk 1]
```diff
@@ -323,12 +334,16 @@ def _kpi_ratio_to_percentage(self, area_name):
                 "total_energy_produced_wh": area_kpis["total_energy_produced_wh"],
                 "total_self_consumption_wh": area_kpis["total_self_consumption_wh"],
                 "base_case_cost": area_kpis.get("base_case_cost"),
+                "utility_bill": area_kpis.get("utility_bill"),
+                "fit_revenue": area_kpis.get("fit_revenue"),
```
Comment (faizan2590) [inline gsy_framework/sim_results/kpi.py:338]: One of the actual fix

File: gsy_framework/sim_results/kpi.py [hunk 2]
```diff
@@ -368,7 +384,19 @@ def restore_area_results_state(self, area_dict: Dict, last_known_state_data: Dic
...
+        if area_dict["uuid"] not in self.savings_state:
+            if not has_grand_children(area_dict):
+                self.savings_state[area_dict["uuid"]] = SavingsKPI()
+                self.savings_state[area_dict["uuid"]].base_case_cost = (
+                    last_known_state_data["base_case_cost"])
+                self.savings_state[area_dict["uuid"]].utility_bill = (
+                    last_known_state_data["utility_bill"])
+                self.savings_state[area_dict["uuid"]].fit_revenue = (
+                    last_known_state_data["fit_revenue"])
+                self.savings_state[area_dict["uuid"]].d3a_cost = (
```
Comment (faizan2590) [inline gsy_framework/sim_results/kpi.py:396]: One of the actual fix.

File: gsy_framework/sim_results/kpi.py [hunk 3]
```diff
@@ -368,7 +384,19 @@ def restore_area_results_state(self, area_dict: Dict, last_known_state_data: Dic
                 last_known_state_data["total_self_consumption_wh"]
             self.state[area_dict["uuid"]].self_consumption_buffer_wh = \
                 last_known_state_data["self_consumption_buffer_wh"]
-
+        if area_dict["uuid"] not in self.savings_state:
+            if not has_grand_children(area_dict):
```
Comment (Hassan754) [inline gsy_framework/sim_results/kpi.py:388]:
```suggestion
        if not (area_dict["uuid"] in self.savings_state or has_grand_children(area_dict)):
```

**Review by Hassan754 — APPROVED** [review]
LGTM, can you please create a ticket to refactor and create unit tests for the simulation results handlers?

**Review by spyrostz — APPROVED** [review]
LGTM once other comments are resolved

File: gsy_framework/sim_results/kpi.py (hunk 3 above)
Comment (Hassan754) [reply gsy_framework/sim_results/kpi.py:388]: Update: Nit, can be skipped

Comment (faizan2590) [reply gsy_framework/sim_results/kpi.py:388]: Will merge it now, and drop a comment to address in follow up ticket to refactor and add unittest

**Comment by faizan2590** [comment]
> LGTM, can you please create a ticket to refactor and create unit tests for the simulation results handlers?

[Done!](https://gridsingularity.atlassian.net/browse/GSYE-42?atlOrigin=eyJpIjoiMDIyZjAxMjRiZDhkNGExNWFmM2QzMGFiMWIzODgwMTEiLCJwIjoiamlyYS1zbGFjay1pbnQifQ)

**Pull request merged**